waspc/src/Wasp/AppSpec/App/Dependency.hs

```diff
@@ -13,7 +13,7 @@ data Dependency = Dependency
   { name :: String,
     version :: String
   }
-  deriving (Show, Eq, Data)
+  deriving (Show, Eq, Ord, Data)
```

Comment:
Are we sure how is Ord derived here? I am guessing it is derived in such fashion that it first compares by `name` and then by `version`. But I feel a bit uncomfortable that we have to assume this is so.
Maybe we should instead implement our own instance of Ord, so that we are sure it checks first by name and then by version? That way we are also robust to somebody adding fields to this record, or them changing order, and similar.
Yeah, I would feel much better knowing that we have a stable Ord algorithm here, and not that it depends on the order and number of fields in the record.

Reply:
I looked it up when I used it, and my understanding is that Records are really lists, and that derived `Ord` therefore uses the order in which fields are defined. 

The main risk is that someone were to reorganize the order of the fields making the `name` not the first entry. It would silently accept that and break other behavior (though the tests would likely catch it). So I think you're right and an explicit Ord is safer.